### ISSUE TITLE
Speek shows nothing (totally blank)

### ISSUE THREAD

**Issue by RaidenMK** [Speek-App/Speek#29]
![Speek](https://user-images.githubusercontent.com/59111778/181694583-1c730858-a52a-405e-9e2a-cde3084176e2.png)

I have now tried the speek chat by downloading speek version 1.6.0 directly from github,but as you see on that photo,speek doesn't show anything at all,as in nothing at all. Only thing it says is that Im logged in. So I tried to download Speek again,but this time from Microsoft Store,but even then I still get the same. You need to fix that,otherwise its useless to use it. I cant use the pre-release of speek version 1.6.1 or 1.7.0,because there is none for Windows 11 64bit.

I'm using Windows 11 Home 64bit

* Note: The speek you see is of version 1.6.0

**Comment by Speek-App** [comment]
Thanks or opening an issue!

If I'm not mistaken, then this issue has already been fixed. You can find a pre-release with the fix [here](https://github.com/Speek-App/Speek/releases/tag/v1.6.1-pre-release)

**Comment by RaidenMK** [comment]
Well. I would had used the pre-release if it had a installer for Windows 11 Home 64bit .. there is no installer for Windows 11 for the pre-release. You would had to make an installer for windows 11 home 64bit for me to use it

**Comment by Speek-App** [comment]
Sorry for that. I added a [new release](https://github.com/Speek-App/Speek/releases/tag/v1.7.0-release) which also includes a Windows installer. The Windows Store version follows shortly.

**Comment by RaidenMK** [comment]
Just 1 question when it comes to groups. For me to join a group I created.. Do I have to add a user first,then add them to the group to be able to join the group I made myself? because it seems I cant join the group I created unless I add one to it.. Thats what I think. Could you explain me alittle about it how Groups works?

**Comment by RaidenMK** [comment]
Im trying to open the group like a chatroom,but it seems I cant seem to get the room to be open. I installed speek on my phone as well and added myself and created a group/room and added the member to it,but it seems that it wont open the room

**Comment by Speek-App** [comment]
Thanks for pointing the issues out!
I hope I don't misunderstand you, but regarding the first one, you actually have to also invite yourself (your main windows speek account) to the group. The group itself can't write messages, therefore only the users panel.

**Comment by RaidenMK** [comment]
Can you explain more about that please? like step by step from when you create a group to adding someone? and if you dont mind.. Can I add you on speek or if you have any test "account" on speek or something. because the way to create group/room and how to be in that group confuses me

**Comment by Speek-App** [comment]
No problem at all. But we definitely have to revise the process a bit to make it more intuitive. Especially as the current user isn't added automatically (we should add a checkbox when creating a group).

So first you have to click on Group in the main context menu, which should open a window with all your current groups and an "Add new group" button in the bottom right. Click on that button and the group creation window should open.

![Screenshot from 2022-07-31 10-29-53](https://user-images.githubusercontent.com/98053512/182017963-57e94de9-df53-4e69-b86c-a2a1c1b79d85.png)

Move all the contacts you would like to add to the right and enter a unique group name and an invite message (as seen in the screenshot).

Next, click "Create Group", which should open the launch window of the group where you can add, if needed, proxy or bridge settings etc. Though in most cases, you can just click "Launch Speek with default settings".

![Screenshot from 2022-07-31 10-30-48](https://user-images.githubusercontent.com/98053512/182018092-649e7ef5-58c9-424d-b21e-5454b105f306.png)

Now the group host window should open.

![Screenshot from 2022-07-31 10-31-50](https://user-images.githubusercontent.com/98053512/182018137-75ebc3c0-1c56-413f-8a08-6bad7e884ad6.png)

Here you can click in the main menu on "Add Group Member" to add yourself or another account to the group.

![Screenshot from 2022-07-31 10-33-15](https://user-images.githubusercontent.com/98053512/182018198-02a85e66-299a-4d27-9bde-aec15d6c9c60.png)

This invited account should now receive an invite; accept it to join the group.

![Screenshot from 2022-07-31 10-34-07](https://user-images.githubusercontent.com/98053512/182018254-2b267597-891f-4980-ae7b-673c8bb108bd.png)

Please let me know if something's unclear or maybe not working as intended.

**Comment by RaidenMK** [comment]
- Here is how I did it:
1.) I created an account on speek on android (same on speek for PC ofcourse)
2.) Then I added myself by adding my own speek-ID for android to my contact list for speek on PC. then I added myself by adding my own speek-ID for PC and added myself on speek for android and accepted request for both.
3.) Then I clicked on the 3-lined at the top left of speek on PC and clicked "Groups" and it opened a window and gave myself a groupname with invite message etc,then I clicked "Add" to add my speek-ID for android to my group and then clicked "Create group" and then clicked "Launch Speek.Chat with default settings". After that,it comes up a window called "Speek Group Host" with my androids Speek-ID in the list.
4.) Once number 1 to 3 is done,then I click the 3-lined at the top left of the window called "Speek Group Host" and click "View Speek Group ID" and then scan the group Qr-code with my android to add that group on my android. Then I accept the request via speek on android and its done.
5.) Then finally I copy the speek group-ID and click "add contact" via speek on PC and add the ID as contact. Not sure if its the right way or right steps I did. But I got an error.

- This is the error when the "group" gets online (by doing the 5 steps above):
{"pinned_message":"","total_group_member":2,"users_online":2}

- And this is the error I get when I post a message on via speek on android to the group:
"message":"<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.0//EN\" \"http://www.w3.org/TR/REC-html40/strict.dtd\">\n<html><head><meta name=\"qrichtext\" content=\"1\" /><style type=\"text/css\">\np, li { white-space: pre-wrap; }\n</style></head><body style=\" font-family:'Noto Sans'; font-size:13px; font-weight:400; font-style:normal;\">\n<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">Ok</p></body></html>","name":"Jaxx-Android","id":"<ID>"}

Note: I removed the ID and replaced the Jaxx-Android ID with "<ID>" just in case. But those are the errors I get.

**Comment by Speek-App** [comment]
You should only send invites **from** the group host. Don't send requests to it (the group host won't see them nor can they be accepted -> in order to prevent spam). 

So in your case (which I would consider a bug), you have added the "group" as a "normal user" as a group can't be added this way, hence the weird messages. After that, the group sent an invite to the user, but the user had already been added as a normal user to the phone, which wasn't changed to a group. (I hope that makes sense.)

So that's definitely not ideal and we should make it impossible to even send a request to the group.

**Comment by RaidenMK** [comment]
i know you already sent a guide how to. but if you dont mind. Could you get on this chat?: https://cryptodog.github.io/cryptodog/

Just enter the nickname you want in the "nickname" field and the name called Speek in the "conversation name" field and click connect. I'll be waiting there.

**Comment by Speek-App** [comment]
> but the group I made doesnt show up in the list of groups and thats via speek on PC

That's weird. Have you tried to close and reopen the group list window?

**Comment by RaidenMK** [comment]
Yes i have. I have even deleted the group and recreate it,but it still the same

**Comment by Speek-App** [comment]
So you sent an invite to your speek id on windows and received that invite but the group isn't showing on the left?

**Comment by RaidenMK** [comment]
> So you sent an invite to your speek id on windows and received that invite but the group isn't showing on the left?

Exactly. Normally the group should show up in the same place as the users,but under the category Group since its a group,but the group isnt showing at all

**Comment by Speek-App** [comment]
> > So you sent an invite to your speek id on windows and received that invite but the group isn't showing on the left?
> 
> Exactly. Normally the group should show up in the same place as the users,but under the category Group since its a group,but the group isnt showing at all

But you received the invite? I'm going to try that on windows and see if I can reproduce that.

**Comment by RaidenMK** [comment]
> > > So you sent an invite to your speek id on windows and received that invite but the group isn't showing on the left?
> > 
> > 
> > Exactly. Normally the group should show up in the same place as the users,but under the category Group since its a group,but the group isnt showing at all
> 
> But you received the invite? I'm going to try that on windows and see if I can reproduce that.

Check the previous post I did and you see a photo that it doesnt show the group

**Comment by Speek-App** [comment]
Ok, just tried it on Windows 10 and it worked without a problem. Did you really receive the invite to the group? 

btw. Speek on Android still doesn't supports the creation of groups.

**Comment by RaidenMK** [comment]
> Ok, just tried it on Windows 10 and it worked without a problem. Did you really receive the invite to the group?
> 
> btw. Speek on Android still doesn't supports the creation of groups.

well. if it works on yours,then it must be working on my side. I just dont get it why I cant see the group.

But then I just wonder. Where and how do I see the group invite on PC version of speek? because I couldnt see anything. Im really confused on that part. Besides Im using Windows 11 Home 64bit.

**Comment by Speek-App** [comment]
Just to make sure, you've sent an invite from the group host window (the small one with only the contact list) to your speek account on windows (your normal account)?

**Comment by RaidenMK** [comment]
> Just to make sure, you've sent an invite from the group host window (the small one with only the contact list) to your speek account on windows (your normal account)?

well. first I added myself (the speek id on my android) and added it to my contacts list on speek on PC,then I created a group and invited that android user (aka myself) to that group and then I clicked "create group". normally when you do that,the group should come up in the list under category "group",but it doesnt

**Comment by Speek-App** [comment]
Now I understand. That's actually the issue I mentioned in the small explanation. You have to (at least with the current version) invite yourself to the group afterwards. It's not automatically adding your account to the group.

**Comment by RaidenMK** [comment]
> Now I understand. That's actually the issue I mentioned in the small explanation. You have to (at least with the current version) invite yourself to the group afterwards. It's not automatically adding your account to the group.

Just to be sure its done correctly,since it works on your side.. Could you make a video with step by step from when you create a account,until you invite a user to making a group until add a user to the group? it would probably be alot easier (I hope)..

**Comment by Speek-App** [comment]
It should be (at least roughly) explained from the third image onwards. Basically, you just have to send an invite from the group host window to your main speek id on windows.

**Comment by RaidenMK** [comment]
> It should be (at least roughly) explained from the third image onwards. Basically, you just have to send a normal invite with the "Add Group Member" context menu button from the group host window to your main speek id on windows. ![Screenshot from 2022-07-31 14-06-13](https://user-images.githubusercontent.com/98053512/182025582-a1ff85fc-081c-4fd9-a021-80754220fab4.png)

The only user I added to my contact list via speek on PC,is my own speek ID on my android.. then I clicked "Add group Member" and added my own speek-ID for android to the group on PC,like you showed me now and earlier.. But still the group doesnt show up on the list.. it only shows my own speek-ID that my android has.. thats all. No group whatsoever is showing

**Comment by Speek-App** [comment]
> added my own speek-ID for android to the group
You should add the speek id of your windows account (the one where you would like the group to appear). Hope I'm not misunderstanding you though.

**Comment by RaidenMK** [comment]
> > added my own speek-ID for android to the group
> 
> You should add the speek id of your windows account (the one where you would like the group to appear). Hope I'm not misunderstanding you though.

Sorry for all the trouble. it just confused. But I got the group to show now. my own 2 ID's were in the group already.. I just had to remove them both from the group and add them back somehow into the group..

**Comment by Speek-App** [comment]
> > > added my own speek-ID for android to the group
> > 
> > 
> > You should add the speek id of your windows account (the one where you would like the group to appear). Hope I'm not misunderstanding you though.
> 
> Sorry for all the trouble. it just confused. But I got the group to show now. my own 2 ID's were in the group already.. I just had to remove them both from the group and add them back somehow into the group..

I'm glad you got it working.
Sorry again for the still a bit unintuitive UI. We will definitely improve the UI for making groups. Thanks a lot for pointing out those flaws.

> oh yeah,i forgot. Do you have any own group that u share with people that people can join? if so.. it would be nice if there was one.

We currently still don't have a group but will maybe make one after the 1.7 release.

> Almost forgot. When you are in a group,it show some kind of encryption thingy in paranthesis beside ur nickname.. Is there a way to make it so it doesnt show and only show the nick?

That's the hash of the speek id of the user (to not make the actual speek id of the user known to the other members of the group). That's a nice to have as the members can always change their nickname and could otherwise impersonate another one. However, there is currently no way to disable it, but if there's a need, we can add an option to do so.
